src/fit.jl

```diff
@@ -152,7 +152,7 @@ function learn_constraint!(bbc::BlackBoxClassifier, ignore_feas::Bool = false; k
     check_sampled(bbc)
     set_param(bbc, :reloaded, false) # Makes sure that we know trees are retrained. 
     lnr = base_classifier()
-    IAI.set_params!(lnr; classifier_kwargs(; kwargs...)...)
+    IAI.set_params!(lnr; minbucket = length(bbc.vars) + 1, classifier_kwargs(; kwargs...)...)
```

Comment:
This makes sure that the minbucket depends on the number of variables involved...